pkg/v1/random/image.go

```diff
@@ -141,3 +124,32 @@ func (i *image) LayerByDiffID(diffID v1.Hash) (partial.UncompressedLayer, error)
 	}
 	return l, nil
 }
+
+// Layer returns a layer with pseudo-randomly generated content.
+func Layer(byteSize int64, fileName string, mt types.MediaType) (v1.Layer, error) {
+	var b bytes.Buffer
```

Comment:
We can get the digest using `io.MultiWriter` while we `CopyN` it, and avoid buffering.